Handle replicator instance start time during upgrades better

nickva opened the pull request:
During cluster upgrades from 3.2 to 3.3 when instance start time switched from being always  `0` to an actual timestamp, replication jobs will crash when endpoints are upgraded. Replication jobs were started when endpoint emitted a `0` and then it becomes a non-`0` value which will crash the next checkpoint attempt.

After the crash jobs will restart and continue fine where they left off without rewinding. However they will make a logging mess while they crash. All four workers will exit with the `{checkpoint_commit_failure,...}` error. This commit make it the checkpoint ignore mismatches if one of the instance start times is 0.

Issue: https://github.com/apache/couchdb/issues/4621

big-r81 reviewed (approved):
+1

A small personal preference: I find the previous variable names more clearer and readable than the new short ones...

nickva commented:
@big-r81 good idea, I thought it looked a bit smaller and better with shorter variables it's not a lot better. Reverted back to original names. Also added a comment indicated what is happening with the `<<"0">>` check there and why we're doing it.
